app/forms/mappings/Formatters.scala

```diff
@@ -123,11 +123,12 @@ trait Formatters {
         finalResult <- resultWithoutCommas match {
                          case s if s.matches(DecimalRegexp) =>
                            Seq(FormError(key, doNotUseDecimalsKey, args)).asLeft[Int]
-                         case s if Try(s.toLong).isFailure => Seq(FormError(key, nonNumericKey, args)).asLeft[Int]
-                         case s if s.toLong < 1 =>
-                           Seq(FormError(key, outOfRangeKey, Seq(1, ONE_BILLION))).asLeft[Int]
-                         case s if s.toLong > ONE_BILLION =>
-                           Seq(FormError(key, outOfRangeKey, Seq(1, ONE_BILLION))).asLeft[Int]
+                         case s if Try(s.toLong).isFailure && !s.matches(WholeNumber) =>
```

Comment:
Why do we need this ```matches(WholeNumber)```?

Reply:
To distinguish between nonNumericKey error and outOfRangeKey error. Regex confirms that it's an actual numeric value.

Reply:
If no match is present for example "100000000000000000000000000" returns nonNumericKey error.